src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/TradesMigrationService.cs

```diff
@@ -140,52 +95,133 @@ TimeSpan sqlTimeout
 
             _tradesMigrationHealthService.State = TradesMigrationState.Finished;
         }
+        
+        public class TradesProcessor
+        {
+            private readonly TradesMigrationHealthService _healthService;
+            private readonly ICandlesHistoryRepository _historyRepo;
+            private readonly Dictionary<CandleTimeInterval, ICandle> _activeCandles;
+            private readonly string _assetPairId;
+            private readonly int _persistenceQueueMaxSize;
 
-        #endregion
+            // ReSharper disable once MemberCanBePrivate.Global
+            public readonly Dictionary<CandleTimeInterval, List<ICandle>> PersistenceCandleQueue; // It's better to make it private, but we need to unit test the class.
```

Comment:
Extract separate TradesMigrationPersistenceQueue and incapsulate this member there. It should use TImerTrigger to periodicaly check in separate thread, if PersistenceCandleQueue is filled enough. This will drastically increase overall performance of migration, since new trades reading and already performed candles writing could will be made in parallel. FlushCandlesIfAny could be removed then, since TradesMigrationPersistenceQueue just writes all that it's has. See CandlesPersistenceQueue as example. 

And you don't need to expose PersistenceCandleQueue as public member for tests, you could just make mock of ITradesMigrationPersistenceQueue in your tests

Of course you need to use concurrent-safe collection in this case.